sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -2610,7 +2635,7 @@ object GpuOverrides {
         }),
     exec[CollectLimitExec](
       "Reduce to single partition and apply limit",
-      ExecChecks(TypeSig.commonCudfTypes + TypeSig.DECIMAL, TypeSig.all),
+      ExecChecks(TypeSig.all - TypeSig.UDT, TypeSig.all),
```

Comment:
I think this would allow an array of UDT through, we need to make sure that UDT is removed from nested as well.

Reply:
Oh and SortExec should be updated to match these too.